data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/config/PipelineParserConfiguration.java

```diff
@@ -7,14 +7,22 @@
 
 import com.amazon.dataprepper.model.plugin.PluginFactory;
 import org.opensearch.dataprepper.parser.PipelineParser;
+import org.opensearch.dataprepper.parser.PipelineParser;
+import org.opensearch.dataprepper.peerforwarder.PeerForwarder;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
 public class PipelineParserConfiguration {
 
     @Bean
-    public PipelineParser pipelineParser(final DataPrepperArgs dataPrepperArgs, final PluginFactory pluginFactory) {
-        return new PipelineParser(dataPrepperArgs.getPipelineConfigFileLocation(), pluginFactory);
+    public PipelineParser pipelineParser(
+            final DataPrepperArgs dataPrepperArgs,
+            final PluginFactory pluginFactory,
+            final PeerForwarder peerForwarder
```

Comment:
This approach is working very nicely. The `PeerForwarder` is created elsewhere and that is the only class that really needs to come out of that package into `PipelineParser`. This will help `PeerForwarder` vary without requiring changes to `PipelineParser`.